### PR TITLE
DashboardService: Consolidate constructors

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Working on #31866, I realized that `DashboardService` has two different constructors. This becomes especially a problem in #31866, since I need a dependency on the SQLStore service (due to global state falling out of use). Therefore, I'm making this PR to consolidate `DashboardService`'s constructors into sharing code. Also removing unnecessary code (thanks @marefr).

### CODE REVIEW

File: pkg/api/dashboard.go [hunk 1]
```diff
@@ -122,7 +122,7 @@ func (hs *HTTPServer) GetDashboard(c *models.ReqContext) response.Response {
 		meta.FolderUrl = query.Result.GetUrl()
 	}
 
-	provisioningData, err := dashboards.NewProvisioningService().GetProvisionedDashboardDataByDashboardID(dash.Id)
+	provisioningData, err := dashboards.NewService(hs.DataService).GetProvisionedDashboardDataByDashboardID(dash.Id)
```
Comment (marefr) [inline pkg/api/dashboard.go:125]: Can you please keep the `NewProvisioningService` factory method that returns a `DashboardProvisioningService`? This function can call `NewService`. This way we're only exposing the dashboard provisioning interface in the provisioning package and has a smaller foot-print and doesn't risk that provisioning calling the non-provisioning methods on the dashboard service. 

The same reasoning goes for someone having a dependency on the dashboard service shouldn't be able to call provisioning methods. This was the historic reasoning for keeping DashboardService and DashboardProvisioningService separate.

Comment (aknuds1) [reply pkg/api/dashboard.go:125]: @marefr OK, will do.

File: pkg/services/dashboards/dashboard_service.go [hunk 2]
```diff
@@ -33,19 +33,17 @@ type DashboardProvisioningService interface {
 	DeleteProvisionedDashboard(dashboardId int64, orgId int64) error
 }
 
-// NewService factory for creating a new dashboard service
+// NewService is a factory for creating a new dashboard service.
```
Comment (marefr) [inline pkg/services/dashboards/dashboard_service.go:36]: Looked at this again. I must have miss something. I cannot see that the `reqHandler` field on `dashboardServiceImpl` is being used anywhere?

Comment (aknuds1) [reply pkg/services/dashboards/dashboard_service.go:36]: @marefr good catch, thanks. There isn't any need for it in this PR, only in the originating PR.

**Review by wbrowne — APPROVED** [review]
LGTM

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**